lib/src/client.dart

```diff
@@ -1810,8 +1860,19 @@ class Client extends MatrixApi {
       onSyncStatus.add(SyncStatusUpdate(SyncStatus.error,
           error: SdkError(exception: e, stackTrace: s)));
       if (e.error == MatrixError.M_UNKNOWN_TOKEN) {
-        Logs().w('The user has been logged out!');
-        await clear();
+        final onSoftLogout = this.onSoftLogout;
```

Comment:
In theory this can happen on more than just sync (basically any request), but sync would be the most common one. Should we add this logic everywhere (we already have a global error handler for example), or is this good enough for now?

Reply:
I would vote for having this here as handling it here will also make sure that it pauses the sync correctly. If we handle it somewhere else, sync loop does not know when to restart